muxtools/subtitle/font.py

```diff
@@ -143,8 +192,115 @@ def clean_name(fontname: str) -> str:
+                    
+                    except Exception as e:
+                        danger(f"Failed to subset font '{fontname}' (possibly corrupt/invalid font): {log_escape(str(e))}", collect_fonts)
+                    
+                    else:
+                        # It's important that the family name and font name are both replaced.
+                        # Usually, the family name is used in styles, but sometimes the full font name is used.
+                        
+                        # Insertion order is important as well, to avoid partial replacements.
+                        # For example:
+                        # - old_family_name = "Arial"
+                        # - old_font_name = "Arial Bold"
+                        # If we replaced "Arial" first, then "Arial Bold":
+                        # - \fnArial Bold -> \fnArialSubset Bold
+                        # which would not match the new name.
+                        # We need to replace the one which is longer first.
+                        
+                        if len(old_font_name) > len(old_font_family_name):
+                            fonts_to_be_replaced[old_font_name] = new_font_name
+                            fonts_to_be_replaced[old_font_family_name] = new_font_family_name
+                        else:
+                            fonts_to_be_replaced[old_font_family_name] = new_font_family_name
+                            fonts_to_be_replaced[old_font_name] = new_font_name
+
+                        old_size = os.path.getsize(outpath)
+                        new_size = os.path.getsize(new_path)
+                        
+                        info(f"Subsetted font '{fontname}' ({len(characters_set)} glyphs, {sizeof_fmt(old_size)} -> {sizeof_fmt(new_size)})", collect_fonts)
+
+                        outpath = new_path
+
+            
+            found_fonts.append(MTFontFile(outpath))
+    
+    # Update additional subfiles with new font names
+    # We do this at the end to avoid multiple reads/writes
+    if fonts_to_be_replaced:
+        for sub_obj in subset_all_subfiles:
+            doc = sub_obj._read_doc()
+
+            for event in doc.events:
+                for old_font_name, new_font_name in fonts_to_be_replaced.items():
+                    event.text = event.text.replace('\\fn' + old_font_name, '\\fn' + new_font_name)
+            
+            for style in doc.styles:
+                for old_font_name, new_font_name in fonts_to_be_replaced.items():
+                    if style.fontname == old_font_name:
+                        style.fontname = new_font_name
+                
+            sub_obj._update_doc(doc)
+
+            info(f"Updated font names in subfile '{sub_obj.file.name}'", collect_fonts)
+
+    #for r in ["*.[tT][tT][fF]", "*.[oO][tT][fF]", "*.[tT][tT][cC]", "*.[oO][tT][cC]"]:
+    #    for f in get_workdir().glob(r):
+    #        found_fonts.append(MTFontFile(f))
```

Comment:
stylistic argument: old, unused code should not be left around commented; version control exists for that purpose